### PR TITLE
Display warning when viewing a commit isn't contained in a branch or a tag.

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

**Comment by Gusted** [comment]
Should such warning also be included when you browse a /tree from a specific commit(which IIRC was also "vulnerable" to this)

**Comment by zeripath** [comment]
The commit page already shows which branch things belong to.

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -2950,3 +2950,6 @@ error.probable_bad_default_signature = "WARNING! Although the default key has th
 unit = Unit
 error.no_unit_allowed_repo = You are not allowed to access any section of this repository.
 error.unit_not_allowed = You are not allowed to access this repository section.
+
+[commit]
+warn.no_branch_or_tag = This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
```
Comment (Gusted) [inline options/locale/locale_en-US.ini]:
```suggestion
warn.no_branch_or_tag = This commit does not belong to any branch in this repository, and may belong to a fork outside of the repository.
```

**Comment by lunny** [comment]
> The commit page already shows which branch things belong to.

Yes, but a warning is also needed I think.

**Comment by zeripath** [comment]
https://try.gitea.io/arandomer/kanna/commit/c1136e41fffabe422994947ba7d3d3a7c8f263ff

Screenshot from 2022-01-30 10-36-08

https://try.gitea.io/arandomer/pathological/commit/77d4598bde3c0b8b142b03eed0acac13a4c0a121

Screenshot from 2022-01-30 10-36-41


https://try.gitea.io/GitUser0123/123/commit/52e2a71928db929556445dc492a9af1f16ca9f14

Screenshot from 2022-01-30 10-37-10

---

So this is already being calculated and we should not be recalculating this.

We should be using the already calculated branch information and then displaying things a bit more clearly.

**Comment by denyskon** [comment]
@lunny Are you still interested in this PR? If so, could you update the branch and resolve conflicts?

**Comment by lunny** [comment]
> @lunny Are you still interested in this PR? If so, could you update the branch and resolve conflicts?

I have enough time to maintain this PR.
Anyone would like to pick this one? Then I can close it.

**Comment by delvh** [comment]
Given the new architecture, that is basically implemented already, you simply need to adapt the frontend to display a warning in that case.
I think that's easier than resolving the conflicts of this PR.

**Pull request closed without merging**